src/client/Microsoft.Identity.Client/Internal/Requests/RequestBase.cs

```diff
@@ -104,9 +108,35 @@ public async Task<AuthenticationResult> RunAsync(CancellationToken cancellationT
                     AuthenticationRequestParameters.RequestContext.Logger.ErrorPii(ex);
                     throw;
                 }
+                finally
+                {
+                    TrackTelemetryIfNecessary(
```

Comment:
Since we have 2 telemetry systems, I suggest using more descriptive naming (or comments) to differentiate.